src/audio/audio_manager.cpp

```diff
@@ -16,17 +21,74 @@ AudioManager::AudioManager() : muted(true) {
         if (!music.openFromFile(filename)) {
             std::cerr << "AudioManager: Cannot open " << filename << std::endl;;
         }
-        music.setLoop(true);
-        if (!muted)
+
+        music.setVolume(d.volume);
+        music.setLoop(d.loop);
+
+        if (!muted) {
```

Comment:
If we return immediately, we won't be setting the volume and looping. Say we enter the build phase as muted and decided to un-mute halfway. Then the volume would be at the default 100 and it won't loop. This is mainly due to how the mute toggle works. When we un-mute we just call play on the music.
Here, even when the music is muted, we just make sure everything is set up correctly so that when the player decides to un-mute, the music would just start playing with the right settings..